### PR TITLE
Added API test for custom uploaded content capsule sync (BZ1340686)

### PR DESCRIPTION
**Pull request by abalakh**
https://bugzilla.redhat.com/show_bug.cgi?id=1340686
```python
py.test -v tests/foreman/api/test_contentmanagement.py -k test_positive_uploaded_content_library_sync
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /Users/andrii/workspace/env/bin/python
Added API test for custom uploaded content capsule sync (BZ1340686)
cachedir: .cache
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.18.1, services-1.2.1, mock-1.6.2, cov-2.5.1
collected 7 items
2017-09-28 17:38:44 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/api/test_contentmanagement.py::CapsuleContentManagementTestCase::test_positive_uploaded_content_library_sync PASSED

============================== 6 tests deselected ==============================
=================== 1 passed, 6 deselected in 54.15 seconds ====================
```

### CODE REVIEW

File: tests/foreman/api/test_contentmanagement.py [hunk 1]
```diff
@@ -131,6 +138,76 @@ def update_capsule_download_policy(self, capsule_id, download_policy):
         proxy.update(['download_policy'])
 
     @tier4
+    def test_positive_uploaded_content_library_sync(self):
+        """Ensure custom repo with no upstream url (manually uploaded content)
+        published to Library is syncing to capsule automatically
```
Comment (ldjebran) [inline tests/foreman/api/test_contentmanagement.py]:
```
Ensure that custom repository with no upstream url and manually uploaded content when published to Library is synchronized to external capsule automatically
```
what do you think?

File: tests/foreman/api/test_contentmanagement.py [hunk 2]
```diff
@@ -131,6 +138,76 @@ def update_capsule_download_policy(self, capsule_id, download_policy):
...
+
+        :id: f5406312-dd31-4551-9f03-84eb9c3415f5
+
+        :BZ: 1340686
+
+        :expectedresults: publishing custom repo with no upstream url triggers
+            capsule sync and custom content is present on capsule afterwards
```
Comment (ldjebran) [inline tests/foreman/api/test_contentmanagement.py]:
```
:expectedresults:  custom content is present on external capsule
```

File: tests/foreman/api/test_contentmanagement.py [hunk 3]
```diff
@@ -131,6 +138,76 @@ def update_capsule_download_policy(self, capsule_id, download_policy):
...
+        # Publish new version of the content view
+        cv.publish()
+        cv = cv.read()
+        self.assertEqual(len(cv.version), 1)
+        # Assert that a task to sync lifecycle environment to the capsule
+        # is started (or finished already)
+        sync_status = capsule.content_get_sync()
+        self.assertTrue(
+            len(sync_status['active_sync_tasks']) >= 1 or
+            sync_status['last_sync_time']
```
Comment (ldjebran) [inline tests/foreman/api/test_contentmanagement.py]: more readable 
```python
     (len(sync_status['active_sync_tasks']) >= 1 
      or sync_status['last_sync_time'])
```

**Comment by abalakh** [comment]
@ldjebran @oshtaier addressed

**Pull request merged**